src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -114,6 +116,22 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         }
     }
 
+    hstring Command::ID() const noexcept
+    {
+        return hstring{ _ID };
+    }
+
+    bool Command::GenerateID()
+    {
+        auto actionAndArgsImpl{ winrt::get_self<implementation::ActionAndArgs>(_ActionAndArgs) };

```

Comment:
will this ever be null?

Reply:
These commands came from `_ActionMap`, and those can [never point](https://github.com/microsoft/terminal/blob/5f3a857192b2fa6f4e5ae87eb3ee59a1571515a2/src/cascadia/TerminalSettingsModel/ActionMap.cpp#L79) to `nullptr`, but I've added a check just in case